### PR TITLE
Fix test sending body as url parameter

### LINKED ISSUE #65718: ClientYamlTestClient can send too-long requests
Our YAML tests will randomly try and send a request body as a URL-encoded set of query parameters instead.  There is a hard limit on how long an HTTP query string can be, and so the method `ClientYamlTestClient#sendBodyAsSourceParam` checks the size of the body and aborts if it is too long.  However, this check does not take into account URL encoding, which means that a request body containing a large number of characters that need to be encoded can end up being longer than the HTTP query string limit.

### PR DESCRIPTION
**Pull request by nik9000**
Our test framework randomly sends the body of requests over the `source`
parameter. We never send the body if it is more than 2000 bytes becuse
our HTTP receiver can't handle lines more the 4096 bytes. The thing is,
when we do that 2000 bytes check we do it against the reported length of
body, not the body after it has been url encoded. Than url encoding
strings can *vastly* increase their size. Which could cause us to send
some request over the URL that are longer than 4096 bytes.

This fixes that by checking the url encoded length as well. We keep the
2000 byte check of the unencoded length because it is a nice fast check,
even if it is a bit inaccurate.

Closes #65718

### CODE REVIEW

File: test/framework/src/main/java/org/elasticsearch/test/rest/yaml/ClientYamlTestClient.java [hunk 1]
```diff
@@ -158,8 +166,8 @@ public ClientYamlTestResponse callApi(String apiName, Map<String, String> params
             }
             String contentType = entity.getContentType().getValue();
             //randomly test the GET with source param instead of GET/POST with body
-            if (sendBodyAsSourceParam(supportedMethods, contentType, entity.getContentLength())) {
-                logger.debug("sending the request body as source param with GET method");
+            if (sendBodyAsSourceParam(supportedMethods, contentType, entity)) {
+                logger.error("sending the request body as source param with GET method");
```
Comment (romseygeek) [inline test/framework/src/main/java/org/elasticsearch/test/rest/yaml/ClientYamlTestClient.java:170]: Should this be changed back to `debug`?

File: rest-api-spec/src/main/resources/rest-api-spec/test/search/30_limits.yml [hunk 2]
```diff
@@ -116,9 +116,6 @@ setup:
 
 ---
 "Regexp length limit":
-  - skip:
-      version: all
-      reason: Long regex breaks HTTP query length when request body is sent as a query param - https://github.com/elastic/elasticsearch/issues/65718
```
Comment (romseygeek) [inline rest-api-spec/src/main/resources/rest-api-spec/test/search/30_limits.yml:121]: NB on backport this needs to restore a previous skip version that looks like this:
```
 - skip:
      version: " - 6.99.99"
      reason: "The regex length limit was introduced in 7.0.0"

```

**Review by romseygeek — APPROVED** [review]
LGTM - one question and one comment but no need for another review.

File: test/framework/src/main/java/org/elasticsearch/test/rest/yaml/ClientYamlTestClient.java (hunk 1 above)
Comment (nik9000) [reply test/framework/src/main/java/org/elasticsearch/test/rest/yaml/ClientYamlTestClient.java:170]: yeah.

File: rest-api-spec/src/main/resources/rest-api-spec/test/search/30_limits.yml (hunk 2 above)
Comment (nik9000) [reply rest-api-spec/src/main/resources/rest-api-spec/test/search/30_limits.yml:121]: Thanks!

**Pull request merged**